src/kv/genericserialisewrapper.h

```diff
@@ -76,22 +76,31 @@ namespace kv
     W private_writer;
     W* current_writer;
     Version version;
+    bool has_contents = false;
 
     std::shared_ptr<AbstractTxEncryptor> crypto_util;
 
     // must only be set by set_current_domain, since it affects current_writer
     SecurityDomain current_domain;
 
     template <typename T>
-    void serialise_internal(T&& t)
+    void serialise_internal(T&& t, bool content_update = true)
     {
```

Comment:
Adding a branch on every single serialise_internal is unfortunate, is there another way to do this?

Reply:
As discussed I will either go back to checking the size of the serializer for `1` or remove the changes from `GenericSerialiseWrapper` all together and have the `serialise()` function decide if each serializer has actual data or not.